detector.go

```diff
@@ -86,26 +186,35 @@ func (r *Require) hasTopLevelVersions() bool {
 	return r.Version != ""
 }
 
+type depKey struct {
+	name  string
+	mixin bool
+	any   bool
```

Comment:
Can these both be defined at the same time (and what would that mean if so)? Just curious...

Reply:
Yes, a stackpack could present itself as being able to install _any_ mixin. This would be what an `apt-buildpack` might do. This way it can pass detection and resolve any other mixin request from downstream buildpacks without having to statically (or even dynamically) present every single option it could do.

Reply:
Ohhh... I see... so if `any` is true then we'd expect `mixin` to always be true, right? (Do we do any check for this?)